combined_robot_hw/src/combined_robot_hw.cpp

```diff
@@ -0,0 +1,238 @@
+// POSSIBILITY OF SUCH DAMAGE.
+//////////////////////////////////////////////////////////////////////////////
+
+#include <algorithm>
+#include "combined_robot_hw/combined_robot_hw.h"
+
+namespace combined_robot_hardware
+{
+  CombinedRobotHW::CombinedRobotHW() :
+    robot_hw_loader_("hardware_interface", "hardware_interface::RobotHW")
+  {}
+
+  bool CombinedRobotHW::init(ros::NodeHandle& root_nh, ros::NodeHandle &robot_hw_nh)
+  {
+    root_nh_ = root_nh;
+    robot_hw_nh_ = robot_hw_nh;
+
+    std::vector<std::string> robots;
+    if (!robot_hw_nh.getParam("robot_hardware", robots)) {return false;}
+
+    std::vector<std::string>::iterator it;
+    for (it = robots.begin(); it != robots.end(); it++)
+    {
+      if (!loadRobotHW(*it))
+      {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  bool CombinedRobotHW::prepareSwitch(const std::list<hardware_interface::ControllerInfo>& start_list,
+                             const std::list<hardware_interface::ControllerInfo>& stop_list)
+  {
+    // Call the prepareSwitch method of the single RobotHW objects.
+    std::vector<boost::shared_ptr<hardware_interface::RobotHW> >::iterator robot_hw;
+    for (robot_hw = robot_hw_list_.begin(); robot_hw != robot_hw_list_.end(); ++robot_hw)
+    {
+      std::list<hardware_interface::ControllerInfo> filtered_start_list;
+      std::list<hardware_interface::ControllerInfo> filtered_stop_list;
+
+      // Generate a filtered version of start_list and stop_list for each RobotHW before calling prepareSwitch
+      filterControllerList(start_list, filtered_start_list, *robot_hw);
+      filterControllerList(stop_list, filtered_stop_list, *robot_hw);
+
+      if (!(*robot_hw)->prepareSwitch(filtered_start_list, filtered_stop_list))
+        return false;
+    }
+    return true;
+  }
+
+  void CombinedRobotHW::doSwitch(const std::list<hardware_interface::ControllerInfo>& start_list,
+                        const std::list<hardware_interface::ControllerInfo>& stop_list)
+  {
+    // Call the doSwitch method of the single RobotHW objects.
+    std::vector<boost::shared_ptr<hardware_interface::RobotHW> >::iterator robot_hw;
+    for (robot_hw = robot_hw_list_.begin(); robot_hw != robot_hw_list_.end(); ++robot_hw)
+    {
+      std::list<hardware_interface::ControllerInfo> filtered_start_list;
+      std::list<hardware_interface::ControllerInfo> filtered_stop_list;
+
+      // Generate a filtered version of start_list and stop_list for each RobotHW before calling doSwitch
+      filterControllerList(start_list, filtered_start_list, *robot_hw);
+      filterControllerList(stop_list, filtered_stop_list, *robot_hw);
+
+      (*robot_hw)->doSwitch(filtered_start_list, filtered_stop_list);
+    }
+  }
+
+  bool CombinedRobotHW::loadRobotHW(const std::string& name)
+  {
+    ROS_DEBUG("Will load robot HW '%s'", name.c_str());
+
+    ros::NodeHandle c_nh;
+    // Constructs the robot HW
+    try{
```

Comment:
roscpp style guide: "Braces, both open and close, go on their own lines (no "cuddled braces")."